build.gradle

```diff
@@ -3,14 +3,17 @@ plugins {
     id "org.checkerframework" version "0.6.33"
 }
 
-group = 'org.example'
-version = '1.0-SNAPSHOT'
+group = 'edu.njit.jerse'
+version = '0.1-SNAPSHOT'
 
 repositories {
     mavenCentral()
 }
 
 dependencies {
+    // TODO: this project has other dependencies (e.g., the Checker Framework itself). Some of these
+    // dependencies are stored as .jar files in the lib/ directory. Instead, all the dependencies should
+    // be managed by the build system directly.
     testImplementation platform('org.junit:junit-bom:5.9.1')
```

Comment:
We are using the .jar files in lib/ to run the Checker Framework once the targeted method is minimized. My assumption is that we can run the CF after Specimin from a gradle file, I just do not know how at the moment. If you have any suggestions, I would love to hear them.

Reply:
Invoking the CF programmatically might be tricky. What I would do is something like the following:
* add the CF as a dependency in the Gradle file (i.e., here). That way it will be tracked appropriate by Gradle.
* somehow force Gradle to actually download the CF's jar from Maven Central before running ASHE. I'm not exactly sure how to do that, but it might be that having it as a dependency (even if it's not actually used in the Java code) is enough. Or, we might need to explicitly make "download the CF" a dependency of some gradle task. We'll have to experiment a bit with this.
* when gradle downloads a dependency, it puts that dependency in a predictable location under `~/.m2/repository/`. We can use that to actually find the appropriate CF version in the code that we use to invoke it. Note that the version number is in the path, so we'll need to make sure we account for that, too.

Can you open an issue for doing this? I don't think it's urgent, but I don't want us to forget about it.

Reply:
@kelloggm Thanks for the response. Please see `[Issue #13](https://github.com/jonathan-m-phillips/ASHE_Automated-Software-Hardening-for-Entrypoints/issues/13)`. We could add more detail to it, but it was created to fix the issues discussed in this conversation.